### PR TITLE
Introduce 'USE_COMPILER_DEFAULT_LIBC' cmake option for switching from jerry-libc to a libc, provided by compiler

### PR DESCRIPTION
**Pull request by ruben-ayrapetyan**

### CODE REVIEW

**Comment by seanshpark** [comment]
Good to me

**Comment by egavrin** [comment]
Good to me

File: Makefile [hunk 1]
```diff
@@ -69,10 +69,18 @@
...
  # List of include paths for external libraries (semicolon-separated)
   EXTERNAL_LIBS_INTERFACE ?=
  # External libc interface
-  EXTERNAL_LIBC_INTERFACE ?= $(shell pwd)/third-party/nuttx/include
+  ifeq ($(USE_COMPILER_DEFAULT_LIBC),YES)
+   ifneq ($(EXTERNAL_LIBC_INTERFACE),)
+    $(error EXTERNAL_LIBC_INTERFACE should not be specified in case compiler's default libc is used)
+   endif
+  else
+   EXTERNAL_LIBC_INTERFACE ?= $(shell pwd)/third-party/nuttx/include
```
Comment (LaszloLango) [inline Makefile]: I think after #330 this isn't needed anymore.

Comment (ruben-ayrapetyan) [inline Makefile]: Yes, seems so. I'll rebase to current master.

Comment (ruben-ayrapetyan) [inline Makefile]: Done.

**Comment by LaszloLango** [comment]
lgtm

**Comment by egavrin** [comment]
`make push`

**Pull request merged**